stackrox/stackrox
# ROX-18523: Delete use of tippy.js via ui-components HoverHint

## Description

Prerequisite to remove ui-components as a dependency.

These are leftovers that were not included in #4987

### Analysis

PatternFly 4 depends on tippy.js 5.1.2 which is previous major version. PatternFly 5 might upgrade to 6.3.7 or maybe omit the dependency.

HoverHint and DetailedTooltipOverlay used in Vulnerability Management or Compliance:

1. src/Components/visuals/LabeledBarGraph/LabeledBarGraph.js
    * src/Containers/VulnMgmt/widgets/MostCommonVulnerabilities.js

2. src/Components/visuals/Scatterplot/Scatterplot.js
    * src/Containers/VulnMgmt/widgets/TopRiskyEntitiesByVulnerabilities.js

3. src/Components/visuals/HorizontalBarChart.js
    * src/Components/visuals/SunburstDetailSection.js obsolete because conditional rendering only if `units === 'percentage'` but Sunburst renders it with `units="value"` prop.
    * src/Containers/Compliance/widgets/StandardsAcrossEntity.js deletion fixes a mistake I made in #6854

4. src/Components/visuals/VerticalBarChart.js
    * src/Containers/Compliance/widgets/EntityCompliance.js does not provide `hint` for HoverHint

5. src/Components/visuals/VerticalClusterBar.js
    * src/Containers/Compliance/widgets/StandardsByEntity.js

### PatternFly design guidelines

HoverHint attaches the tooltip via DOM methods on mouseover for bars and points. Although PatternFly `Tooltip` element, which uses tippy.js package, might have similar implementation, it encloses the element that is the subject of the tooltip. But the bars and points are svg elements rendered by react-viz package.

https://www.patternfly.org/v4/components/tooltip/design-guidelines

> Use tooltips to provide additional information on a data point or element in a chart or table.

So far, so good. All of the occurrences are in bar chart or scatterplot graph.

https://www.patternfly.org/v4/components/tooltip/design-guidelines#content-considerations

> Keep your tooltips clear and concise. Use the fewest number of words without sacrificing meaning.

Some of these tooltips have extensive information. All of the bars or points are links to pages that include the information.
![vulnerability-management](https://github.com/stackrox/stackrox/assets/11862657/6f595da8-5690-4019-b4d0-8b2b001d42f0)

> Tooltips should provide new and valuable information. Never use a tooltip to repeat information already available in the UI.

Look at it the other way around. The most concise tooltips contain information that could and should be in UI.
![compliance](https://github.com/stackrox/stackrox/assets/11862657/9c45d72b-3494-408a-ba0c-a2b6016ad5f0)

For example, imagine **Standards across clusters** as a table:

| Standard            | Failing controls | Passing controls |
| :---                |             ---: |             ---: |
| CIS Docker v1.2.0   |                0 |             100% |
| CIS Kubernetes v1.5 |               15 |              64% |
| HIPAA 164           |                9 |              73% |
| NIST SP 800-190     |               12 |              68% |
| NIST SP 800-53      |               11 |              29% |
| PCI DSS 3.2.1       |               16 |              90% |

The table could even have a fourth column for a horizontal bar.

### Solution

Move files from src/Components/visuals to src/Containers/Whichever/widgets so it is obvious to delete the parts with their counterparts.

Delete `HoverHint` and `DetailedTooltipOverlay` because:
1. Some of the content contracts PatternFly design guidelines.
2. Compliance and Vulnerability Management are in process of being superseded, although slowly.
3. It is difficult, if even possible, to replace classic `HoverHint` with PatternFly `Tooltip` element. Adding insult to injury, example of replacements have bad-looking scroll bars, even if not needed:
    ![PillTooltipBody](https://github.com/stackrox/stackrox/assets/11862657/90aa0547-04d3-4327-b622-7edb0d83e6a7)

### Residue

Because bundle size **increased** when I deleted tippy.js as a dependency in apps/platform/package.json even though yarn.lock did not change, I will defer that until:
1. Delete use of `Avatar` and `Message` elements.
2. Remove dependency on ui-components.
3. Remove references to ui-components from build configuration.

## Checklist

## Testing Performed

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 4 = 4231488 - 4231484
        total: -3259 = 10246230 - 10249489
    * `ls -al build/static/js/*.js | wc -l`
        0 = 80 - 80 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/compliance: see no tooltips on bars
2. Visit /main/vulnerability-management: see no tooltips on points nor bars

### Integration testing

* compliance/complianceDashboard.test.js
* vulnmanagement/dashboard.test.js
* vulnmanagement/dashboardToEntityPage.test.js
